src/vmm/src/persist.rs

```diff
@@ -280,6 +268,29 @@ fn snapshot_memory_to_file(
     }
 }
 
+/// Validate the microVM version and translate it to its corresponding snapshot data format.
+pub fn snapshot_version_sanity_check(
+    version: &Option<String>,
+    version_map: &VersionMap,
+    device_manager: &MMIODeviceManager,
+) -> std::result::Result<u16, CreateSnapshotError> {
+    use self::CreateSnapshotError::InvalidVersion;
+    match version {
+        Some(version) => match FC_VERSION_TO_SNAP_VERSION.get(version) {
```

Comment:
This looks so confusing now. Can you rewrite this to get rid of the nested matches ?

Reply:
Done.